pkg/agent/controllers/agent/agent.go

```diff
@@ -0,0 +1,205 @@
+package agent
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+)
+
+const (
+	handlerResync          = time.Hour * 2
+	submarinerIpamGlobalIp = "submariner.io/globalIp"
+)
+
+var excludeSvcList = map[string]bool{"kubernetes": true, "openshift": true}
+
+func New(spec *LighthouseAgentSpecification, config InformerConfigStruct) (*Controller, error) {
+	exclusionNSMap := map[string]bool{"kube-system": true, "submariner": true, "openshift": true,
+		"submariner-operator": true, "kubefed-operator": true}
+	for _, v := range spec.ExcludeNS {
+		exclusionNSMap[v] = true
+	}
+	agentController := &Controller{
+		kubeClientSet:    config.KubeClientSet,
+		serviceWorkqueue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "Services"),
+		servicesSynced:   config.ServiceInformer.Informer().HasSynced,
+
+		excludeNamespaces: exclusionNSMap,
+	}
+
+	klog.Info("Setting up event handlers")
+	config.ServiceInformer.Informer().AddEventHandlerWithResyncPeriod(cache.ResourceEventHandlerFuncs{
+		AddFunc: agentController.enqueueService,
+		UpdateFunc: func(old, newObj interface{}) {
+			agentController.handleUpdateService(old, newObj)
+		},
+		DeleteFunc: agentController.handleRemovedService,
+	}, handlerResync)
```

Comment:
The resync period is already set to 60 sec on the Informer - any particular reason to override that here?

Reply:
I am still working on this, was refactoring stuff and trying out different values. Will use the one on informer.